xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -202,7 +206,39 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
                 references.pop();
             }
         }
-
+        
+        // Indent the heading Level
+        if (parameters.getIndentHeadingLevel() > 0 && result.getChildren().size() > 0
+                && result.getParent() instanceof SectionBlock) {
+
+            // Getting all the HeaderBlocks
+            List<HeaderBlock> headingList = result.getBlocks(new ClassBlockMatcher(HeaderBlock.class),
+                    Block.Axes.DESCENDANT_OR_SELF);
+
+            if (!headingList.isEmpty()) {
+                HeaderLevel[] headerLevels = { HeaderLevel.LEVEL1, HeaderLevel.LEVEL2, HeaderLevel.LEVEL3,
+                        HeaderLevel.LEVEL4, HeaderLevel.LEVEL5, HeaderLevel.LEVEL6 };
+                HeaderLevel headinglevel;
+
+                // Looping through all the HeaderBlocks and changing levels of LEVEL1 HeaderBlocks
+                for (HeaderBlock heading : headingList) {
+                    if (heading.getLevel() == HeaderLevel.LEVEL1) {
```

Comment:
That doesn't feel right: you are not supposed to only change the levels for the Level1, but actually for all headers.